doc/man3/BIO_printf.pod

```diff
@@ -40,8 +40,9 @@ buffer is too small.
 
 =head1 NOTES
 
-Both BIO_snprintf() and BIO_vsnprintf() null terminate their output when there
-is insufficient space.
+Except when I<n> is 0, both BIO_snprintf() and BIO_vsnprintf() terminate
+their output with B<'\0'> when there is insufficient space to output
```

Comment:
I'd use `C<'\0'>`.  Also I'd add `even` or `also`. So it won't confuse anyone to think that the NUL is not added in some other cases.

Reply:
I have a suspicion that there are cases where it isn't added.  There are a lot of potential error paths for extremely unlikely / impossible things.  For example, any call to `doapr_outch` could result in failure, many aren't possible but ruling them out would be a mammoth undertaking.

The only reasonable way I think we can guarantee termination in all cases would be to zero the entire buffer up front.

Reply:
I've changed to C<>.

Reply:
It is not that hard. We do not care about the returns for failing `ossl_assert()` as those should be internal errors only. And in case of BIO_snprintf/vsnprintf() the buffer is NULL as we do not do any allocations. So doapr_outch() should not really fail.

Reply:
I see only two return 0 calls in fmtfp() that might be relevant (i.e. they are not failures from doapr_outch()). Perhaps we should add there `doapr_outch(sbuffer, buffer, &currlen, maxlen, '\0');` before the `return 0;`?

Reply:
Assuming the asserts don't happen, you seem to be correct.  I've made the three changes in `fmtfp()` and slightly updated the wording.  I'm not entirely comfortable turning this into an API guarantee given the convoluted nature of this code, but we'll see...